src/options/options.js

```diff
@@ -1,3 +1,188 @@
+    shortcutClearBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+// Reset button handler
+shortcutResetBtnArray.forEach((item) => {
+  const shortcutResetBtnElement = item;
+  shortcutResetBtnElement.addEventListener('click', () => {
+    const currentInput = shortcutResetBtnElement.parentNode.firstChild.nextSibling;
+    const getDefaultShortCutsById = defaultShortCutsArr.find((el) => el.id === currentInput.id);
+    currentInput.value = getDefaultShortCutsById.shortcut;
+    browser.commands.update({ name: currentInput.id, shortcut: currentInput.value });
+    browser.storage.local.set({ storedShortCutsArr: [{ id: currentInput.id, shortcut: currentInput.value }] });
+    shortcutResetBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+function getOS() {
+  const { userAgent } = window.navigator;
+  const platform = window.navigator?.userAgentData?.platform || window.navigator.platform;
+  const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
+  const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
+  const iosPlatforms = ['iPhone', 'iPad', 'iPod'];
+
+  if (macosPlatforms.indexOf(platform) !== -1) {
+    return 'Mac OS';
+  } if (iosPlatforms.indexOf(platform) !== -1) {
+    return 'iOS';
+  } if (windowsPlatforms.indexOf(platform) !== -1) {
+    return 'Windows';
+  } if (/Android/.test(userAgent)) {
+    return 'Android';
+  } if (/Linux/.test(platform)) {
+    return 'Linux';
+  }
+
+  return null;
+}
+const isMac = getOS() === 'Mac OS';
+
+const normalizeKey = (key, keyCode) => {
+  const alphabet = /^([a-z]|[A-Z])$/;
+  if (alphabet.test(key)) return key.toUpperCase();
+
+  const digit = /^[0-9]$/;
+  const func = /^F([0-9]|1[0-2])$/;
+  const homes = /^(Home|End|PageUp|PageDown|Insert|Delete)$/;
+  if (digit.test(key) || func.test(key) || homes.test(key)) return key;
+
+  const space = /^\s$/;
+  if (space.test(key)) return 'Space';
+
+  const arrows = /^(ArrowUp|ArrowDown|ArrowLeft|ArrowRight)$/;
+  if (arrows.test(key)) return key.split('Arrow')[1];
+
+  const medias = /^(MediaPlayPause|MediaStop)$/;
+  if (medias.test(key)) return key;
+  if (key === 'MediaTrackNext') return 'MediaNextTrack';
+  if (key === 'MediaTrackPrevious') return 'MediaPrevTrack';
+
+  const keyCode0 = 48;
+  if (keyCode0 <= keyCode && keyCode <= keyCode0 + 9) return keyCode - keyCode0;
+
+  if (keyCode === 188) return 'Comma';
+  if (keyCode === 190) return 'Period';
+
+  return '';
+};
+
+function handleKeyDown(e) {
+  if (e.repeat) return;
+  if (e.key === 'Tab') {
+    window.document.activeElement.blur();
+    return;
+  }
+  const normalizedKey = normalizeKey(e.key, e.keyCode);
+  let error = e.target.parentNode.nextElementSibling.innerText = '';
+  const mediaKeys = /^(MediaPlayPause|MediaStop|MediaNextTrack|MediaPrevTrack)$/;
+  const funcKeys = /^F([0-9]|1[0-2])$/;
+  const modifierKeys = /^(Control|Alt|Shift|Meta)$/;
```

Comment:
why you do it here but not in normalizeKey function

Reply:
This is necessary to display different types of error messages.